tests/network/probes.go

```diff
@@ -92,50 +97,57 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 				serverStarter(vmi, readinessProbe, 1500)
 			} else {
 				By(specifyingVMReadinessProbe)
-				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
-				vmi.Spec.ReadinessProbe = readinessProbe
-				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
+				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(
+					newRandomFedoraVMIWithGuestAgent(withReadinessProbe(readinessProbe)))
```

Comment:
I agree with @jean-edouard , just because that nasty looking launcher function with `f()()` look. It is hard to see what is what, everything seems obfuscated to me.

Reply:
@EdDev I don't follow - you'll have `f()()` never the less. That comes from the `tests.VMILauncherIgnoreWarnings` function definition, not my usage of it.

Reply:
I now rechecked the code, and apparently I'm using 2 lines everywhere else. For the sake of consistency (and humoring you ...) I am following @jean-edouard 's suggestion.